services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchSpec.groovy

```diff
@@ -24,4 +33,118 @@ class SwitchSpec extends BaseSpecification {
         def exc = thrown(HttpClientErrorException)
         exc.rawStatusCode == 400
     }
+
+    def "Delete nonexistent switch"() {
+        given: "Nonexistent switch ID"
+        def switchId = new SwitchId("de:ad:be:ef:99:99:99:99")
+
+        when: "Try to delete nonexistent switch"
+        northboundService.deleteSwitch(switchId, false)
+
+        then: "Got 404 NotFound"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+    }
+
+    def "Delete active switch"() {
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        when: "Try to delete active switch"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch must be deactivated"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+    }
+
+    def "Delete inactive switch with ISLs"() {
+        requireProfiles("virtual")
+
+        given: "An active switch"
+        def sw = topology.getActiveSwitches()[0]
+
+        and: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(sw.getDpId())
+
+        when: "Try to delete inactive switch with ISLs"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has ISLs"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+
+        and: "Cleanup: Activate the switch back"
+        lockKeeperService.reviveSwitch(sw.getDpId())
+    }
+
+    def "Delete inactive switch with single flow"() {
+        requireProfiles("virtual")
+
+        given: "An active switch with flow"
+        def sw = topology.getActiveSwitches()[0]
+        def flow = flowHelper.singleSwitchFlow(sw)
+        northboundService.addFlow(flow)
+        assert Wrappers.wait(WAIT_OFFSET) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Deactivate this switch"
+        lockKeeperService.knockoutSwitch(sw.getDpId())
+
+        when: "Try to delete inactive switch with flow"
+        northboundService.deleteSwitch(sw.getDpId(), false)
+
+        then: "Got 400 BadRequest because switch has flow"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
```

Comment:
It is not clear whether it is not removed due to existing flow or due to active ISLs being still present on the switch. In order to check that the reason is indeed the existing flow we can 
1. Verify the error message in case if it differs for isls/flows cases. It should state the 'flow' reason, not 'isl'
or
2. Eliminate the ISL factor by forcing all ISLs on this switch to down state.

same for the two below tests with flows

Reply:
I'll try to do it, but 
1. For some reasons `exc` has only "400 " error message. (Real http request returns correct error messages)
2. ISL in down state is not enough. I need to remove ISL from DB. At this moment I'm trying to find a way to remove and restore ISL in DB.

Reply:
Found the updates in #328 . In this case you'll have to delete ISL from DB. We don't have such method atm, but this should be an easy fix in  org.openkilda.testing.service.database.DatabaseNeoImpl

Reply:
1. You should be able to do `exc.message == "whatever"`

Reply:
`exc.responseBodyAsString` helped! thanks